DemoApp/controller/user.controller.js

```diff
@@ -35,17 +35,15 @@ const userRegister=async(req,res,employeetype)=>{
 
         //create new user
         const HashPassword=await bcrypt.hash(userDetails.password,12)
-        const NewUser=new User({
-        // ...userDetails,
-        name : req.body.name,
-        username : req.body.username,
-        email: req.body.email,
-        employeetype:employeetype,
-        phone: req.body.phone,
-        totalexperience : req.body.totalexperience,
-        password:HashPassword,
-        
-    });
+        const UserObject={
+          name : req.body.name,
+          username : req.body.username,
+          email: req.body.email,
+          employeetype:employeetype,
+          phone: req.body.phone,
+          totalexperience : req.body.totalexperience,
+          password:HashPassword};
+        const NewUser=new User({UserObject});
     console.log("Saved")  
     await NewUser.save();
```

Comment:
You can reduce these lines simply by doing something like this ->
const NewUser = await User({your user object}).save();